Dockerfile

```diff
@@ -1,13 +1,14 @@
 FROM ubuntu/postgres:12-20.04_beta
 
-RUN echo 'APT::Install-Recommends "0";\nAPT::Install-Suggests "0";' > /etc/apt/apt.conf.d/01norecommend \
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+RUN printf 'APT::Install-Recommends "0";\nAPT::Install-Suggests "0";' > /etc/apt/apt.conf.d/01norecommend \
     && apt-get update -y \
     # Install Patroni and its dependencies.
     && apt-cache depends patroni | sed -n -e 's/.* Depends: \(python3-.\+\)$/\1/p' \
             | grep -Ev '^python3-(sphinx|etcd|consul|kazoo|kubernetes)' \
             | xargs apt-get install -y git python3-pip python3-wheel \
-    && pip3 install setuptools \
-    && pip3 install 'git+https://github.com/zalando/patroni.git@v2.1.2#egg=patroni[kubernetes]' \
+    && pip3 install --no-cache-dir setuptools==60.9.3 \
+    && pip3 install --no-cache-dir 'git+https://github.com/zalando/patroni.git@v2.1.2#egg=patroni[kubernetes]' \
```

Comment:
Why `--no-cache-dir` out of curiosity?

Reply:
Hi @marcoppenheimer! It's just a suggestion of the linter to reduce the image size. It actually gives an CI error if we don't provide this flag.